src/Microsoft.TestPlatform.Utilities/CodeCoverageRunSettingsProcessor.cs

```diff
@@ -0,0 +1,484 @@
+            };
+        }
+        #endregion
+
+        #region Public Interface
+        /// <summary>
+        /// Processes the current settings for the code coverage data collector.
+        /// </summary>
+        /// 
+        /// <param name="currentSettings">The code coverage settings.</param>
+        /// 
+        /// <returns>An updated version of the current run settings.</returns>
+        public XmlNode Process(string currentSettings)
+        {
+            if (string.IsNullOrEmpty(currentSettings))
+            {
+                return null;
+            }
+
+            // Load current settings from string.
+            var document = new XmlDocument();
+            document.LoadXml(currentSettings);
+
+            return this.Process(document.DocumentElement);
+        }
+
+        /// <summary>
+        /// Processes the current settings for the code coverage data collector.
+        /// </summary>
+        /// 
+        /// <param name="currentSettingsDocument">
+        /// The code coverage settings document.
+        /// </param>
+        /// 
+        /// <returns>An updated version of the current run settings.</returns>
+        public XmlNode Process(XmlDocument currentSettingsDocument)
+        {
+            if (currentSettingsDocument == null)
+            {
+                return null;
+            }
+
+            return this.Process(currentSettingsDocument.DocumentElement);
+        }
+
+        /// <summary>
+        /// Processes the current settings for the code coverage data collector.
+        /// </summary>
+        /// 
+        /// <param name="currentSettingsRootNode">The code coverage root element.</param>
+        /// 
+        /// <returns>An updated version of the current run settings.</returns>
+        public XmlNode Process(XmlNode currentSettingsRootNode)
+        {
+            if (currentSettingsRootNode == null)
+            {
+                return null;
+            }
+
+            // Get the code coverage node from the current settings. If unable to get any
+            // particular component down the path just add the default values for that component
+            // from the default settings document and return since there's nothing else to be done.
+            var codeCoveragePathComponents = new List<string>() { "CodeCoverage" };
+            var currentCodeCoverageNode = this.SelectNodeOrAddDefaults(
+                currentSettingsRootNode,
+                this.defaultSettingsRootNode,
+                codeCoveragePathComponents);
+
+            if (currentCodeCoverageNode == null)
+            {
+                return currentSettingsRootNode;
+            }
+
+            // Get the code coverage node from the default settings.
+            var defaultCodeCoverageNode = this.ExtractNode(
+                this.defaultSettingsRootNode,
+                Exclusion.BuildPath(codeCoveragePathComponents));
+
+            foreach (var exclusion in this.exclusions)
```

Comment:
You are using exclusions only here so it doesn't make sense to keep reference on instance. You can just before this line generate list of exlusions and yes in this case it can be IEnumerable :)

You are modifying state of exclusions during process so if I understand it correctly creating 1 instance of this Processor and calling in parallel Process can produce issues. Lets keep it simple

Reply:
Done. You're right, it really makes more sense to downgrade from an instance member to a local scoped variable.